packages/connect-sql/src/meta-type.ts

```diff
@@ -0,0 +1,91 @@
+import { Database, RunResult, Statement } from "better-sqlite3";
+import { DBConnection, Store, UploadMetaFnParams } from "./types";
+import { SQLiteConnection } from "./sqlite-adapter";
+
+export interface MetaType {
+    readonly name: string
+    readonly branch: string
+    readonly blob: Uint8Array
+}
+
+export interface MetaRecordKey {
+    readonly name: string
+    readonly branch: string
+}
+
+export interface MetaRecord extends MetaRecordKey {
+    readonly blob: Uint8Array
+    readonly updated_at: Date;
+}
+
+export class MetaSQLRecordBuilder {
+    #name?: string;
+    #branch?: string;
+    #blob?: Uint8Array;
+
+    static fromUploadMetaFnParams(data: Uint8Array, params: UploadMetaFnParams): MetaSQLRecordBuilder {
+        const bld = new MetaSQLRecordBuilder()
+        bld.#name = params.name
+        bld.#branch = params.branch
+        bld.#blob = data
+        return bld
+    }
+
+    build(): MetaRecord {
+        return {
+            name: this.#name!,
+            branch: this.#branch!,
+            blob: Buffer.from(this.#blob!),
+            updated_at: new Date()
+        }
+    }
+
+}
+
+
+
+class SQLiteMetaStore implements Store<MetaRecord, MetaRecordKey> {
+    readonly client: Database;
+    insertStmt?: Statement;
+    selectStmt?: Statement;
+    readonly table: string;
+    constructor(client: Database, table: string = 'metastore') {
+        this.client = client
+        this.table = table
+    }
+    async start(): Promise<SQLiteMetaStore> {
+        await this.client.prepare(`CREATE TABLE IF NOT EXISTS ${this.table} (
+        name TEXT not null,
```

Comment:
we track the CID and parent CIDs of the blob. and when a blob comes in with a parent CID that matches a current blob's CID, it replaces that current blob, which can be deleted. here is [the client side](https://github.com/fireproof-storage/fireproof/blob/main/packages/connect-netlify/src/connect-netlify.ts#L38) and [here is the server](https://github.com/fireproof-storage/fireproof/blob/82daf5a7cbe8d2b37742d35a7d52591db3a27194/packages/connect-netlify/src/server.ts#L29)

Reply:
thx for the explaining it -- but this line seams not related to this.

Reply:
I think the table need a column for cid and a column for parents (short array of cids, don't need anything fancy)

Reply:
but that is for a connector, not a storage, to track multi-writer